java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparser/Navigator.java

```diff
@@ -23,8 +23,34 @@ private Navigator() {
         if (cu.getTypes() == null) {
             return Optional.empty();
         }
-        return cu.getTypes().stream().filter((t) -> t.getName().equals(name)).findFirst();
+        
+        String[] nameParts = name.split("\\.", 2);
+        final String typeName = nameParts[0];
+        
+        Optional<TypeDeclaration> type = cu.getTypes().stream().filter((t) -> t.getName().equals(typeName)).findFirst();
+        
+        if (nameParts.length > 1 && type.isPresent()) {
+            return findType(type.get(), nameParts[1]);
```

Comment:
We could assign nameParts[1] to a variable so the name would make more obvious what it is